crates/velyst_macros/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+use darling::{
+    ast::Data,
+    util::{Flag, Ignored},
+    Error, FromDeriveInput, FromField,
+};
+use proc_macro::TokenStream;
+use quote::{quote, ToTokens};
+use syn::{parse_macro_input, spanned::Spanned, DeriveInput, Ident, Index};
+
+#[derive(FromDeriveInput)]
+#[darling(forward_attrs(typst_path))]
+struct TypstPathArgs {
+    ident: Ident,
+    #[darling(with = "convert_typst_path")]
+    attrs: TypstPathValue,
+}
+
+struct TypstPathValue(syn::Expr);
+
+fn convert_typst_path(attrs: Vec<syn::Attribute>) -> darling::Result<TypstPathValue> {
+    let mut filtered = attrs
+        .into_iter()
+        .filter(|a| a.path().is_ident("typst_path"));
+
+    let attr = filtered
+        .next()
+        .ok_or(darling::Error::custom("#[typst_path] is required"))?;
+
+    if filtered.next().is_some() {
+        return Err(darling::Error::custom("#[typst_path] can only occur once"));
+    }
+
+    let val = attr.meta.require_name_value()?.value.clone();
+
+    Ok(TypstPathValue(val))
+}
+
+/// Macro for deriving `TypstPath`.
+///
+/// Usage
+/// ```rust
+/// # use velyst_macros::TypstPath;
+/// #[derive(TypstPath)]
+/// #[typst_path = "path_to_file.typ"] // Path relative to bevy asset dir
```

Comment:
probably can just be `#[typst_path = "path_to_file.typ"]`?

Tho not sure if it's possible with `darling` crate as I have not used it before.

Reply:
Do you mean leaving out the `#[derive(TypstPath)]`?
This is technically possible, but it would require using an [attribute macro](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) instead of a [derive macro](https://doc.rust-lang.org/reference/procedural-macros.html#derive-macros).
As the name implies, derive macros are mostly intended for deriving traits and as such are simpler than attribute macros. The main difference is that derive macros cannot change the definition of the struct they are defined on, while attribute macros completely replace the element they're defined on.
Derive macros also communicate more clearly what's happening ("the trait TypstPath is being derived") and are very common in the bevy ecosystem, e.g. for deriving `Component` and `Resource`.
That's why I feel like using a derive macro is more appropiate here.

Reply:
That's true. I think it's also better for future if there's other functions we need to implement in the trait.